core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -361,24 +376,31 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
             }
         });
 
-        return mapAcknowledgementFutures(futuresMap);
+        return mapAcknowledgementFutures(futuresMap, Optional.empty());
     }
 
-    private CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData>> mapAcknowledgementFutures(Map<TopicIdPartition, CompletableFuture<Throwable>> futuresMap) {
+    private CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData>> mapAcknowledgementFutures(
+        Map<TopicIdPartition, CompletableFuture<Throwable>> futuresMap,
+        Optional<BiConsumer<Collection<String>, Boolean>> metricsHandler
```

Comment:
Personally, I think it would be neater to have an interface for the metrics handler as opposed to using `BiConsumer<Collection<String>, Boolean>` everywhere.

Reply:
I got rid of 1 BiConsumer by passing BrokerTopicStats to ShareFetch as with recent refactoring we can pass BrokerTopicStats to ShareFetch as the class is moved to `server` module. The othe BiConsumer, I simplified to use Consumer. As there exists fust 1 such functional call now hence skipped creating an interface. Please let me know if you disagree.